code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -378,46 +376,57 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 
 		if(agent_b && air.temperature > 500 && air.toxins > MINIMUM_REACT_QUANTITY )
 			var/datum/gas/oxygen_agent_b/trace_gas = payload.get_or_add_trace_gas_by_type(/datum/gas/oxygen_agent_b)
-			payload.temperature = T0C // Greatly reduce temperature to simulate an endothermic reaction
-			// Itr: .18 Agent B, 20 oxy, 1.3 minutes per iteration, realisticly around 7-8 minutes per crystal.
+			payload.temperature = T0C
 
-			animate_flash_color_fill_inherit(location,"#ff0000",4, 2 SECONDS)
-			playsound(location, 'sound/effects/leakagentb.ogg', 50, 1, 8)
-			if(!particleMaster.CheckSystemExists(/datum/particleSystem/sparklesagentb, location))
-				particleMaster.SpawnSystem(new /datum/particleSystem/sparklesagentb(location))
+			animate_flash_color_fill_inherit(owner,"#ff0000",4, 2 SECONDS)
+			playsound(owner, 'sound/effects/leakagentb.ogg', 50, 1, 8, flags = SOUND_IGNORE_SPACE)
+			if(!particleMaster.CheckSystemExists(/datum/particleSystem/sparklesagentb, owner))
+				particleMaster.SpawnSystem(new /datum/particleSystem/sparklesagentb(owner))
 			trace_gas.moles += 0.18
-			iterations -= 1
-			payload.oxygen = 20
+			molitz.iterations -= 1
+			payload.oxygen = 15
 
 			target.assume_air(payload)
 		else
-			animate_flash_color_fill_inherit(location,"#0000FF",4, 2 SECONDS)
-			playsound(location, 'sound/effects/leakoxygen.ogg', 50, 1, 5)
+			animate_flash_color_fill_inherit(owner,"#0000FF",4, 2 SECONDS)
+			playsound(owner, 'sound/effects/leakoxygen.ogg', 50, 1, 5, flags = SOUND_IGNORE_SPACE)
 			payload.oxygen = 80
-			iterations -= 1
+			molitz.iterations -= 1
 
 			target.assume_air(payload)
 
+
+
 /datum/materialProc/molitz_temp/agent_b
+	max_generations = 1
 	execute(var/atom/location, var/temp)
 		..(location, temp, TRUE)
 		return
 
 /datum/materialProc/molitz_exp
-	var/maxexplode = 1
-	execute(var/atom/location, var/sev)
-		if(maxexplode <= 0) return
-		var/turf/target = get_turf(location)
+	max_generations = 1
+	execute(var/atom/owner, var/sev)
+		if(!istype(owner.material, /datum/material/crystal/molitz))
+			return
+		var/datum/material/crystal/molitz/molitz = owner.material
+		if(molitz.unexploded <= 0)
+			return
+		var/turf/target = get_turf(owner)
 		if(sev > 0 && sev < 4) // Use pipebombs not canbombs!
+			if(molitz.iterations >= 1)
+				playsound(owner, 'sound/effects/leakoxygen.ogg', 50, 1, 5, flags = SOUND_IGNORE_SPACE)
+			if(molitz.iterations == 0)
+				playsound(owner, 'sound/effects/molitzcrumble.ogg', 50, 1, 5, flags = SOUND_IGNORE_SPACE)
 			var/datum/gas_mixture/payload = new /datum/gas_mixture
 			payload.oxygen = 50
 			payload.temperature = T20C
 			target.assume_air(payload)
-			maxexplode -= 1
-			if(owner)
-				owner.setProperty("resonance", 1)
+			molitz.iterations = 2
```

Comment:
Why 2?  Should it be += 2?

Seems like exploding it before using it is a weird trap or am I missing something?

Reply:
The basic idea is that youre releasing all the big pockets of gas but youre fracturing the crystal enough to where the small pockets can leak. Basic idea is they use the sound + the examination text to notice oh i really messed it up and then learn from there